### PR TITLE
DNS Modules

### PR DESCRIPTION
**Pull request by SebastianGode**
Still in development - just for running Tests

### CODE REVIEW

**Comment by SebastianGode** [comment]
@tischrei Integration Test added

File: plugins/modules/dns_floating_ip.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+# Creating a record:
+- name: Creating a record
+  opentelekomcloud.cloud.dns_floating_ip:
+    floating_ip: 123.123.123.123
+    ptrdname: "test2.com."
+    description: "test2nownow"
+    ttl: 300
+    state: present
+
+
```
Comment (anton-sidelnikov) [inline plugins/modules/dns_floating_ip.py:89]: remove extra lines

File: plugins/modules/dns_floating_ip.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      sample: "example.com"
+    status:
+      description: Resource status
+      type: str
+      sample: "ACTIVE"
+    ttl:
+      description: PTR record cache duration (in second) on a local DNS server
+      type: int
+      sample: 300
+
```
Comment (anton-sidelnikov) [inline plugins/modules/dns_floating_ip.py:76]: remove extra lines

File: plugins/modules/dns_zones.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+                zone_email = data[0]['email']
+                zone_check = True
+            else:
+                zone_check = False
+                if self.params['state'] == 'absent':
+                    self.exit(
+                        changed=False,
+                        message=('No Zone found with name: %s' %
+                                 self.params['name'])
+                    )
```
Comment (anton-sidelnikov) [inline plugins/modules/dns_zones.py:190]: you can move that to function

File: tests/integration/targets/dns/tasks/main.yaml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    - name: Set facts
+      set_fact:
+        fl_ip: "{{ fl.floating_ip.floating_ip_address }}"
+        ptrdname: "{{ ( prefix + '.com.' ) }}"
+        description: "{{ ( prefix + 'description' ) }}"
+        zone_name: "{{ ( prefix + '-zone.com.' ) }}"
+        rs_name: "{{ ( prefix + '-rs.' + prefix + '-zone.com.' ) }}"
+
+    - name: Creating a dns_floating_ip entry
+      dns_floating_ip:
```
Comment (anton-sidelnikov) [inline tests/integration/targets/dns/tasks/main.yaml:22]: use fqdn names in tests

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]
comments

**Comment by SebastianGode** [comment]
This should be ready to merge now

**Review by anton-sidelnikov — DISMISSED** [review]

File: tests/integration/targets/dns/tasks/main.yaml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      debug:
+        var: dns_rs.recordset
+
+    - name: assert result
+      assert:
+        that:
+          - dns_rs is success
+          - dns_rs.recordset.description is defined
+
+  always:
```
Comment (tischrei) [inline tests/integration/targets/dns/tasks/main.yaml:132]: ignore_errors: yes is missing for whole always block

File: tests/integration/targets/dns/tasks/main.yaml [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+      debug:
+        var: dns_fl.ptr
+
+    - name: assert result
+      assert:
+        that:
+          - dns_fl is success
+          - dns_fl.ptr.description is defined
+
+    - name: Creating a DNS Zone
```
Comment (tischrei) [inline tests/integration/targets/dns/tasks/main.yaml:56]: Please provide a public and a private Zone scenario.

File: tests/integration/targets/dns/tasks/main.yaml [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+          register: dns_fl_dr
+
+        - name: Dropping DNS Recordset
+          opentelekomcloud.cloud.dns_recordset:
+            zone_id: "{{ dns_zo.zone.id }}"
+            recordset_name: "{{ rs_name }}"
+            state: absent
+          register: dns_rs_dr
+
+        - name: Drop DNS Zone
```
Comment (tischrei) [inline tests/integration/targets/dns/tasks/main.yaml:147]: delete private and public zone

**Review by tischrei — CHANGES_REQUESTED** [review]

File: tests/integration/targets/dns/tasks/main.yaml [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+          - dns_rs.recordset.description is defined
+
+  always:
+    - block:
+        - name: Drop dns_floating_ip entry
+          opentelekomcloud.cloud.dns_floating_ip:
+            floating_ip: "{{ fl_ip }}"
+            state: absent
+          register: dns_fl_dr
+          ignore_errors: true
```
Comment (tischrei) [inline tests/integration/targets/dns/tasks/main.yaml:204]: Please write ignore_errors: true only once for the whole always block. Please see my NAT or DEH tests for this.

**Review by tischrei — CHANGES_REQUESTED** [review]

File: tests/integration/targets/dns/tasks/main.yaml [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+
+    - name: assert result
+      assert:
+        that:
+          - dns_rs is success
+          - dns_rs.recordset.description is defined
+
+  always:
+    - block:
+        git add- name: Drop dns_floating_ip entry
```
Comment (tischrei) [inline tests/integration/targets/dns/tasks/main.yaml:199]: typo

**Review by tischrei — CHANGES_REQUESTED** [review]
typo inside

**Review by tischrei — APPROVED** [review]

File: tests/integration/targets/dns/tasks/main.yaml [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+      set_fact:
+        fl_ip: "{{ fl.floating_ip.floating_ip_address }}"
+        ptrdname: "{{ ( prefix + '.com.' ) }}"
+        description: "{{ ( prefix + 'description' ) }}"
+        zone_public_name: "{{ ( prefix + '-zone.com.' ) }}"
+        zone_private_name: "{{ ( prefix + '-zone.com.' ) }}"
+        rs_name: "{{ ( prefix + '-rs.' + prefix + '-zone.com.' ) }}"
+        network_name: "{{ ( prefix + '-network' )}}"
+        subnet_name: "{{ ( prefix + '-subnet' )}}"
+        router_name: "{{ ( prefix + '-router' )}}"
```
Comment (anton-sidelnikov) [inline tests/integration/targets/dns/tasks/main.yaml:23]: sorry for late comment, but can you use here some suffix in names like dns? it helps to distinguish module when test resources hangs

Comment (tischrei) [reply tests/integration/targets/dns/tasks/main.yaml:23]: don't forget "request_changes" :-D

Comment (SebastianGode) [reply tests/integration/targets/dns/tasks/main.yaml]: @anton-sidelnikov I changed it so that everything has "dns" in it at the end

**Review by anton-sidelnikov — APPROVED** [review]

**Review by tischrei — APPROVED** [review]

**Pull request merged**